DuckDuckGo/TabViewController.swift

```diff
@@ -48,6 +48,15 @@ class TabViewController: WebViewController {
         webEventsDelegate = self
     }
     
+    public var link: Link? {
+        guard let url = url else {
+            return Link.mergeData(primary: nil, secondary: tabModel.link)
+        }
+        
+        let activeLink = Link(title: name, url: url, favicon: favicon)
+        return Link.mergeData(primary: activeLink, secondary: tabModel.link)
+    }
```

Comment:
Given primary will never be nil in this case, `mergeData` could just be an instance method which returns a new Link when merged, e.g.

`return activeLink.merge(with: tabModel.link)`